### PR TITLE
Feat: filter dropdown next to sorting

### PR DESCRIPTION
**Pull request by dmutrokot**

### CODE REVIEW

File: src/components/Filters/components/InputFilter/InputFilter.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+export default function InputFilter({ option, onSetSelectedFilters }) {
+    const classes = styles();
+
+    const [currentValue, setCurrentValue] = useState([]);
+
+    useEffect(() => {
+        setCurrentValue(option.value);
+    }, [option.value]);
+
+    const onInputChange = useCallback((value) => {
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Filters/components/InputFilter/InputFilter.js]: fixed, thanks

File: src/components/Filters/components/MultiButtonSelectionFilter/MultiButtonSelectionFilter.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    }, [option, onSetSelectedFilters, items]);
+
+    return (
+        <div className={classes.wrapper}>
+            <span className={classes.title}>{option.title}</span>
+            <div className={classes.items}>
+                {
+                    items.map((item, index) =>
+                        <Button
+                            className={classNames(classes.item, item.isSelected ? 'selected' : '' )}
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Filters/components/MultiButtonSelectionFilter/MultiButtonSelectionFilter.js]: fixed, thanks

File: src/components/Filters/components/MultiAutocompleteFilter/MultiAutocompleteFilter.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    return (
+        <Autocomplete
+            id={`${option.key}-autocomplete`}
+            multiple
+            value={option.items.filter(item => item.isSelected)}
+            onChange={onHandleAutocompleteChange}
+            options={option.items}
+            getOptionLabel={option => option.title}
+            isOptionEqualToValue={(option, value) => option.value === value.value}
+            renderInput={(params) => (
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Filters/components/MultiAutocompleteFilter/MultiAutocompleteFilter.js]: fixed, thanks

File: src/components/Filters/components/RangeSliderFilter/RangeSliderFilter.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Slider } from '@mui/material';
+
+import styles from './styles';
+import CustomTooltip from 'components/custom/CustomTooltip';
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Filters/components/RangeSliderFilter/RangeSliderFilter.js:4]: fixed, thanks

File: src/components/Filters/components/RangeSliderFilter/RangeSliderFilter.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    useEffect(() => {
+        setCurrentValue(option.value);
+    }, [option.value]);
+
+    const onSliderChange = (value) => {
+        setCurrentValue(value);
+    };
+
+    const onSliderChangeCommited = useCallback((value) => {
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Filters/components/RangeSliderFilter/RangeSliderFilter.js:20]: fixed, thanks

File: src/components/Filters/components/RangeSliderFilter/RangeSliderFilter.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+            <Slider
+                min={option.min}
+                max={option.max}
+                value={currentValue}
+                onChange={(event, value) => onSliderChange(value)}
+                onChangeCommitted={(event, value) => onSliderChangeCommited(value)}
+                valueLabelDisplay='auto'
+                disableSwap
+                size={'small'}
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Filters/components/RangeSliderFilter/RangeSliderFilter.js]: fixed, thanks

File: src/components/Filters/components/SingleAutocompleteFilter/SingleAutocompleteFilter.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        <Autocomplete
+            id={`${option.key}-autocomplete`}
+            disablePortal
+            value={currentValue}
+            getOptionLabel={option => option.title || ''}
+            isOptionEqualToValue={(option, value) => option.isSelected === value.isSelected}
+            options={option.items}
+            disabled={option.items.length === 0}
+            onChange={onHandleChange}
+            renderInput={(params) => (
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Filters/components/SingleAutocompleteFilter/SingleAutocompleteFilter.js]: fixed, thanks

File: src/pages/Client/routes/ClientGotchis.js [hunk 8]
```diff
@@ -52,34 +58,140 @@ const sortings = [
...
+            filter.resetFilterFn(filter);
+        });
+
+        setIsFiltersApplied(false);
+        setCurrentFilters(currentFilters);
+        updateQueryParams(currentFilters);
+    }, [currentFilters, updateQueryParams]);
+
+    const getGotchis = useCallback(() => {
+        return (isSortingChanged || isFiltersApplied) ? modifiedGotchis: gotchis;
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/pages/Client/routes/ClientGotchis.js]: added space, but left brackets for readability

File: src/pages/Client/routes/ClientLendings.js [hunk 9]
```diff
@@ -55,34 +61,140 @@ const sortings = [
...
+            filter.resetFilterFn(filter);
+        });
+
+        setIsFiltersApplied(false);
+        setCurrentFilters(currentFilters);
+        updateQueryParams(currentFilters);
+    }, [currentFilters, updateQueryParams]);
+
+    const getLendings = useCallback(() => {
+        return (isSortingChanged || isFiltersApplied) ? modifiedLendings: lendings;
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/pages/Client/routes/ClientLendings.js]: added space, but left brackets for readability

File: src/pages/GhostExplorer/GhostExplorer.js [hunk 10]
```diff
@@ -1,114 +1,237 @@
...
+            filter.resetFilterFn(filter);
+        });
+
+        setIsFiltersApplied(false);
+        setCurrentFilters(currentFilters);
+        updateQueryParams(currentFilters);
+    }, [currentFilters, updateQueryParams]);
+
+    const getGotchis = useCallback(() => {
+        return (isSortingChanged || isFiltersApplied) ? modifiedGotchis: gotchis;
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/pages/GhostExplorer/GhostExplorer.js]: added space, but left brackets for readability

File: src/pages/Lend/Lend.js [hunk 11]
```diff
@@ -110,36 +132,151 @@ export default function Lend() {
...
+            filter.resetFilterFn(filter);
+        });
+
+        setIsFiltersApplied(false);
+        setCurrentFilters(currentFilters);
+        updateQueryParams(currentFilters);
+    }, [currentFilters, updateQueryParams]);
+
+    const getLendings = useCallback(() => {
+        return (isSortingChanged || isFiltersApplied) ? modifiedLendings: lendings;
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/pages/Lend/Lend.js]: added space, but left brackets for readability

File: src/pages/Lend/Lend.js [hunk 12]
```diff
@@ -96,11 +122,7 @@ export default function Lend() {
 
                 response.forEach((listing) => {
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/pages/Lend/Lend.js]: fixed, thanks

File: src/pages/Lend/Lend.js [hunk 13]
```diff
@@ -160,16 +297,16 @@ export default function Lend() {
                     {/* // !temporary code (hidden feature) */}
                     { linksListView ? (
                         <ol style={{ height: 'calc(100vh - 208px)', overflowY: 'scroll', margin: 0, padding: '10px 0 10px 60px' }}>
-                            {lendings.map((lend) => {
+                            {getLendings().map((lend) => {
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/pages/Lend/Lend.js]: fixed, thanks

File: src/components/SortFilterPanel/SortFilterPanel.js [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+                                    >
+                                        Reset Filters
+                                    </Button>
+                                </div>
+                            </div>
+                        )}
+                    </div>
+            }
+
+            { itemsLength > 0 &&
```
Comment (dudendy) [inline src/components/SortFilterPanel/SortFilterPanel.js:81]: let's be consistent and wrap component into `()`

File: src/components/SortFilterPanel/styles.js [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+        position: 'absolute',
+        left: 8,
+        minWidth: 350,
+        maxWidth: 500,
+        background: theme.palette.background.paper,
+        borderRadius: '4px 0 4px 4px',
+        cursor: 'default',
+        overflow: 'hidden',
+        display: 'none',
+
```
Comment (dudendy) [inline src/components/SortFilterPanel/styles.js:48]: extra space

File: src/pages/Client/Client.js [hunk 16]
```diff
@@ -10,7 +10,6 @@ import styles from './styles';
 export default function Client() {
     const classes = styles();
     const match = useRouteMatch();
-
```
Comment (dudendy) [inline src/pages/Client/Client.js:13]: please add space

**Review by dudendy — CHANGES_REQUESTED** [review]

**Review by dudendy — APPROVED** [review]

**Pull request merged**